lib/langchain/conversation_memory.rb

```diff
@@ -47,7 +43,7 @@ def reduce_messages(exception)
     def context
       return if @context.nil? && @summary.nil?
 
-      [@context, @summary].compact.join("\n")
+      SystemMessage.new([@context, @summary].compact.join("\n"))
     end
 
     private
```

Comment:
We'll need to check the `summarize_messages` method below, I'm not sure what "role" it should have.. maybe it should be a `ChatMessage` with `role: chat`?

Reply:
I checked, and now the difference is that we send `type: [role]` instead of `role: [role]` , but I think we might be good